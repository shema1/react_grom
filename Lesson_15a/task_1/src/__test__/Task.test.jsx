import React from 'react';
import { shallow } from 'enzyme';
import Task from '../Task.jsx.js';

describe('<Task/>', () => {
    it('should display done task', () => {
        const props = {
            text: 'Task1',
            id: 'some-id-1',
            done: true,
            onChange: jest.fn(),
            onDelete: jest.fn()
        }
        const wrappedComponent = shallow(<Task {...props}/>)
        expect(wrappedComponent).toMatchSnapshot();
    });

    it('should display undone task', () => {
        const props = {
            text: 'Task1',
            id: 'some-id-1',
            done: false,
            onChange: jest.fn(),
            onDelete: jest.fn()
        }
        const wrappedComponent = shallow(<Task {...props}/>)
        expect(wrappedComponent).toMatchSnapshot();
    });

    it('should update task on checkbox checked', () => {
        const props = {
            text: 'Task1',
            id: 'some-id-1',
            done: true,
            onChange: jest.fn(),
            onDelete: jest.fn()
        }
        const wrappedComponent = shallow(<Task {...props}/>);
        wrappedComponent.find('.list-item__checkbox').simulate('change');
        expect(props.onChange).toBeCalledWith('some-id-1');
    });

    it('should delete task', () => {
        const props = {
            text: 'Task1',
            id: 'some-id-1',
            done: true,
            onChange: jest.fn(),
            onDelete: jest.fn()
        }
        const wrappedComponent = shallow(<Task {...props}/>);
        wrappedComponent.find('.list-item__delete-btn').simulate('click');
        expect(props.onDelete).toBeCalledWith('some-id-1');
    })
})