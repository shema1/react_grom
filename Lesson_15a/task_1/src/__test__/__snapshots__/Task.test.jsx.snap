// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Task/> should display done task 1`] = `
<li
  className="list-item list-item_done"
>
  <input
    className="list-item__checkbox"
    defaultChecked={true}
    onChange={[Function]}
    type="checkbox"
  />
  <span
    className="list-item__text"
  >
    Task1
  </span>
  <button
    className="list-item__delete-btn"
    onClick={[Function]}
  />
</li>
`;

exports[`<Task/> should display done task 2`] = `
<li
  className="list-item"
>
  <input
    className="list-item__checkbox"
    defaultChecked={false}
    onChange={[Function]}
    type="checkbox"
  />
  <span
    className="list-item__text"
  >
    Task1
  </span>
  <button
    className="list-item__delete-btn"
    onClick={[Function]}
  />
</li>
`;

exports[`<Task/> should display undone task 1`] = `
<li
  className="list-item"
>
  <input
    className="list-item__checkbox"
    defaultChecked={false}
    onChange={[Function]}
    type="checkbox"
  />
  <span
    className="list-item__text"
  >
    Task1
  </span>
  <button
    className="list-item__delete-btn"
    onClick={[Function]}
  />
</li>
`;
